src/Tribe/Service_Providers/Promoter_Connector.php

```diff
@@ -33,5 +33,71 @@ private function hook() {
 		add_filter( 'determine_current_user', tribe_callback( 'promoter.connector', 'authenticate_user_with_connector' ), 20, 1 );
 
 		tribe( 'promoter.pue' );
+
+		// The usage of a high priority so we can push the icon to the end
+		add_action( 'admin_bar_menu', array( $this, 'add_promoter_logo_on_admin_bar' ), 1000 );
+		add_action( 'tribe_common_loaded', array( $this, 'add_promoter_assets' ) );
+	}
+
+	/**
+	 * Add Admin Bar link to the promoter website
+	 *
+	 * @param $wp_admin_bar
+	 */
+	public function add_promoter_logo_on_admin_bar( $wp_admin_bar ) {
+		/** @var Tribe__Promoter__PUE $pue */
+		$pue = tribe( 'promoter.pue' );
+		if ( ! $pue->has_valid_license() ) {
```

Comment:
This method isn't cached, I have concerns about hitting this every admin page load

Reply:
Oh! Right, I think we can replace it with `has_license_key` instead.

Let me know what do you think?

Reply:
cc @sc0ttkclark

Reply:
Yes that sounds much better